zstd.go

```diff
@@ -156,16 +138,31 @@ func Decompress(dst, src []byte) ([]byte, error) {
 	}
 
 	if dst == nil {
-		// x is the 95 percentile compression ratio of zstd on points.mlti payloads
-		dst = make([]byte, len(src)*3)
+		// Attempt to use zStd to determine decompressed size (may result in error or 0)
+		cSize := C.size_t(C.ZSTD_getDecompressedSize(unsafe.Pointer(&src[0]), C.size_t(len(src))))
+
+		if C.ZSTD_isError(cSize) != C.uint(0) {
+			msg := C.ZSTD_getErrorName(cSize)
+			return dst, fmt.Errorf("cannot determine decompressed size: %s", C.GoString(msg))
```

Comment:
Can we just return the `ErrorCode ` here ? (It's also already has the `String()` method)